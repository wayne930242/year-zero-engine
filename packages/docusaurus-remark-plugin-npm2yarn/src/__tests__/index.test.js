/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable no-param-reassign */

import remark from 'remark';
import npm2yarn from '../index';
import vfile from 'to-vfile';
import {join, relative} from 'path';
import mdx from 'remark-mdx';

const staticDir = `./${relative(process.cwd(), join(__dirname, 'fixtures'))}`;

const processFixture = async (name, options) => {
  const path = join(__dirname, 'fixtures', `${name}.md`);
  const file = await vfile.read(path);
  const result = await remark()
    .use(mdx)
    .use(npm2yarn, {...options, filePath: path})
    .process(file);

  return result.toString();
};

describe('npm2yarn plugin', () => {
  test('test: installation file', async () => {
    const result = await processFixture('installation', {
      staticDir,
    });

    expect(result).toMatchSnapshot();
  });

  test('test: plugin file', async () => {
    const result = await processFixture('plugin', {
      staticDir,
    });

    expect(result).toMatchSnapshot();
  });

  test('test: language was not setted', async () => {
    const result = await processFixture('syntax-not-properly-set', {
      staticDir,
    });

    expect(result).toMatchSnapshot();
  });
});
