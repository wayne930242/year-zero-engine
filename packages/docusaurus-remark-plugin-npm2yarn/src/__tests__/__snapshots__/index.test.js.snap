// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm2yarn plugin test: installation file 1`] = `
"import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs defaultValue=\\"npm\\" values={[
    { label: 'npm', value: 'npm', },
    { label: 'Yarn', value: 'yarn', },
  ]}
>
<TabItem value=\\"npm\\">

\`\`\`bash
  $ npm install --global docusaurus
\`\`\`

</TabItem>
<TabItem value=\\"yarn\\">

\`\`\`bash
  $ yarn add --global docusaurus
\`\`\`

</TabItem>
</Tabs>
"
`;

exports[`npm2yarn plugin test: language was not setted 1`] = `
"\`\`\`npm2yarn
npm install --save docusaurus-plugin-name
\`\`\`

\`\`\`bash
npm install --save docusaurus-plugin-name
\`\`\`

\`\`\`shell
npm install --save docusaurus-plugin-name
\`\`\`
"
`;

exports[`npm2yarn plugin test: plugin file 1`] = `
"import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Installing a plugin

A plugin is usually a npm package, so you install them like other npm packages using npm.

<Tabs defaultValue=\\"npm\\" values={[
    { label: 'npm', value: 'npm', },
    { label: 'Yarn', value: 'yarn', },
  ]}
>
<TabItem value=\\"npm\\">

\`\`\`bash
npm install --save docusaurus-plugin-name
\`\`\`

</TabItem>
<TabItem value=\\"yarn\\">

\`\`\`bash
yarn add docusaurus-plugin-name
\`\`\`

</TabItem>
</Tabs>
"
`;
